### DISCUSSION TITLE
Issues Using the Mohr Coulomb with Strain Softening (MCSS) Model in ANURA3D

### DISCUSSION THREAD

**Discussion by GBarbosaS** [Anura3D/Anura3D_OpenSource#280] [Q&A]
Hello, I’m facing difficulties using the Mohr Coulomb with Strain Softening (MCSS) model in ANURA3D. I successfully compiled the program with the MCSS model in both the 2022 and 2024 versions (using a file provided by a friend), but I encountered errors in the command prompt.

Issues Encountered:
Error in 2D Simulation:
During a 2D simulation, I encountered an error related to indexing:
![2D-Error](https://github.com/user-attachments/assets/f20766f0-76a6-4d34-b614-1a807de255a1)
Error in 3D Simulation:
To troubleshoot, I attempted to run the simulation in 3D, but received another error:
![3D-Error](https://github.com/user-attachments/assets/e73118a1-7d76-48f3-adfc-6a297148323d)

I also tried using the files available at this link:
[Anura3D_Moore - VelCorrection Branch](https://github.com/WaveHello/Anura3D_Moore/tree/Anura3D_VelCorrection)
The compilation was successful with these files, but I’m unsure if I used the correct model name or if I made a mistake in Visual Studio.

Even when successfully compiling with the model from the link above, the MCSS model name was not recognized in the .GOM file. I tried several different names, but none worked. (It’s possible I placed the call incorrectly in the code, though I followed the hardcoding manual closely.) However, native models worked perfectly.

Question:
Has anyone faced similar issues with the MCSS model in ANURA3D? Could there be a critical issue in the .GOM file or the code that I might be missing? Any help in identifying the problem would be greatly appreciated!

**Comment by 18052121369** [comment]
@GBarbosaS 
hello,GBarbosaS, I also try to compiled the program with the MCSS model in both the 2022 version or 2024 version, but I met some
questions. CCan you tell me how you compiled successfully ? Or can you share these files ? Perhaps we can solve some problems in the application of MCSS constitutive model together. Here is my email :[wkf18052121369@163.com](url). I will look forward to your reply very much.

**Comment by GBarbosaS** [comment]
@18052121369 I'll explain what I have done so far. I managed to compile the MCSS file in the 2024 version, which is located in this Google Drive folder:
:https://drive.google.com/drive/folders/1K85OtC75zuikmrF1Rhp3UTP6-auKGQ9o?usp=sharing 
The file I am referring to is Velcorrection. I took the model and placed it into the 2024 version, slightly modifying the files described in the following manual:https://github.com/WaveHello/Anura3D-Info/blob/main/Anura3D_HelpGuides/Hard_Coding_ESM_Guide.md.
![image](https://github.com/user-attachments/assets/bf478e6e-3b58-4c8e-8e4a-160595926a16)
I made the calls as follows: his shows where I included the model file, the modification in MPMConvPhase, the model call, the changes in ReadMaterialData.for, and the modifications in ExternalSoilModel.for).
![image](https://github.com/user-attachments/assets/3d61a9b0-cc62-4b9a-86d0-3c78cfbb4f41)
![image](https://github.com/user-attachments/assets/9f4679c2-cc03-4552-a0e3-23bdcc698584)
![image](https://github.com/user-attachments/assets/4632856e-14a5-43bc-a801-c60000dc0cce)
![image](https://github.com/user-attachments/assets/485cbc3f-2ef1-4e90-88da-4eaed8e7cf92)
![image](https://github.com/user-attachments/assets/53d3fab7-118b-4244-95a5-71d282054f1f)
Even so, the program didn’t work with the model, presenting the following error:
Forrtl: severe (157): Program Exception - Access Violation.
I obtained a GID file from a colleague (available in the above drive folder), where I replaced the executable, and the program started running with MCSS. However, it seems unstable and generates errors over time. The test file I used runs each step in 30 seconds with the conventional MohrCoulomb model, but with MCSS, each step took 3 hours and 15 minutes. It eventually presented an error after a few steps, as shown in the following image:
![image](https://github.com/user-attachments/assets/93598ef9-2759-4394-852d-9a2070ec0d28)

**Reply by GBarbosaS** [reply]
Now I am trying to compile the NAMC Model, but I haven’t succeeded yet.

**Reply by 18052121369** [reply]
@GBarbosaS 
Thank you very much for your reply, which is very helpful to me, I will try according to your reply, sincerely thank you
